src/database/sqlite.rs

```diff
@@ -52,6 +52,11 @@ static MIGRATIONS: &[&str] = &[
     "DELETE FROM transactions;",
     "DELETE FROM utxos;",
     "DROP INDEX idx_txid_vout;",
+    "CREATE UNIQUE INDEX idx_utxos_txid_vout ON utxos(txid, vout);",
+    "ALTER TABLE utxos RENAME TO utxos_old;",
+    "CREATE TABLE utxos (value INTEGER, keychain TEXT, vout INTEGER, txid BLOB, script BLOB, is_spent BOOLEAN DEFAULT 0);",
+    "INSERT INTO utxos SELECT value, keychain, vout, txid, script, is_spent FROM utxos_old;",
+    "DROP TABLE utxos_old;",
```

Comment:
To avoid potentially leaving the database in a semi-inconsistent state I would pack all these commands in a single string, so that it counts as a single migration. Plus i would make it into a transaction with `BEGIN/COMMIT` (see: https://www.sqlite.org/lang_transaction.html) so that either it completes fully or fails.